src/Workspaces/CSharp/Portable/Simplification/CSharpInferredMemberNameReducer.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Options;
+
+namespace Microsoft.CodeAnalysis.CSharp.Simplification
+{
+    internal partial class CSharpInferredMemberNameReducer : AbstractCSharpReducer
+    {
+        public override IExpressionRewriter CreateExpressionRewriter(OptionSet optionSet, CancellationToken cancellationToken)
+        {
+            return new Rewriter(optionSet, cancellationToken);
+        }
+
+        private static ArgumentSyntax SimplifyTupleName(ArgumentSyntax node, SemanticModel semanticModel, OptionSet optionSet, CancellationToken cancellationToken)
+        {
+            string inferredName = node.IsParentKind(SyntaxKind.TupleExpression) ?
+                CSharpSimplificationService.ExtractAnonymousTypeMemberName(node.Expression).ValueText :
+                null;
```

Comment:
The name of the method matches the method I copied from compiler. I'd rather keep the name (or avoid copying the code entirely).